neurolib/models/multimodel/builder/base/neural_mass.py

```diff
@@ -0,0 +1,200 @@
+    python_callbacks = []
+
+    # index of noise variable within the full system's input handled by jitcdde
+    noise_input_idx = None
+
+    DESCRIPTION_FIELD = [
+        "index",
+        "name",
+        "mass_type",
+        "num_state_variables",
+        "num_noise_variables",
+        "state_variable_names",
+        "parameters",
+    ]
+
+    def __init__(self, parameters):
+        """
+        :param parameters: parameters of the neural mass
+        :type parameters: dict
+        """
+        assert isinstance(parameters, dict)
+        self.parameters = parameters
+        # used in determining portion of the full system's state vector
+        self.idx_state_var = None
+        self.initialised = False
+
+        # initialise possible helpers
+        self.helper_symbols = {symbol: se.Symbol(symbol) for symbol in self.helper_variables}
+
+        # initialise possible callback functions
+        self.callback_functions = {function: se.Function(function) for function in self.python_callbacks}
+
+        self._validate_parameters()
+
+    def __str__(self):
+        """
+        String representation.
+        """
+        return (
+            f"Neural mass: {self.name} with {self.num_state_variables} "
+            f"state variables: {', '.join(self.state_variable_names)}"
+        )
+
+    def describe(self):
+        """
+        Return description dict.
+        """
+        return {field: getattr(self, field) for field in self.DESCRIPTION_FIELD}
+
+    def _initialize_state_vector(self):
+        """
+        Initialize state vector. By default it is all zeroes.
+        """
+        self.initial_state = [0.0] * self.num_state_variables
+
+    def _validate_parameters(self):
+        """
+        Validate parameters - check if self.parameters contains all required
+        parameters.
+        """
+        assert all(key in self.parameters for key in self.required_parameters), set(self.required_parameters) - set(
+            self.parameters.keys()
+        )
+
+    def _validate_callbacks(self, callback_list):
+        """
+        Validate callbacks - mainly the length and symbolic function names.
+
+        :param callback_list: list of callbacks
+        :type callback_list: list[tuple|list]
+        """
+        assert len(callback_list) == len(self.callback_functions)
+        assert all(
+            callback_from_list[0] == callback_from_init
+            for callback_from_list, callback_from_init in zip(callback_list, self.callback_functions.values())
+        )
+        assert all(callable(callback[1]) for callback in callback_list)
+
+    def init_mass(self, start_idx_for_noise=None):
+        """
+        Initialise neural mass. Usually just initialise the state vector,
+        possibly can be subclassed and do other initialisation.
+        """
+        self._initialize_state_vector()
+        assert self.index is not None
+        if start_idx_for_noise is None:
+            start_idx_for_noise = self.index
```

Comment:
well, in a lot of basic models you have one noise variable per mass, i.e. the noise index in full "noise" matrix (number of masses x time) is the same is mass index within full node / network

Reply:
Mhh not sure if I get it yet (maybe I need to read more of the code): So the noise matrix can be masses x time or masses x noise_per_mass x time then? What would be the value of `start_idx_for_noise` if there were more than one vector per mass? Thanks for your patience with me :)) 🤓

Reply:
OK, I messed up the first explanation :) will start again:
Imagine two cases:
* simple mass with one noise variable, think e.g. excitatory AdEx mass... there is one O-U process as a noise.. in this case, the mass will have `num_noise_variables = 1` so only one index. Now if you have two masses in the node and you wanna simulate two nodes in a network, in total you have 4 masses with one noise variable each, hence 4 noise variables. in this case, the "noise matrix" will be 4 x time. In this particular function then you just set the starting index of noise at the same value as is the index of the mass itself.
* simple mass like Hopf that actually have two noise variables per mass, hence `num_noise_variables = 2`, now it is more tricky because for e.g. 10 masses you actually need to provide 20 noise variables... you can see how this is handled in the Hopf network definition, in particular, the lines:
```
for node_idx, node in enumerate(self.nodes):
            node.init_node(start_idx_for_noise=node_idx * node.num_noise_variables)
```
so each node will start its "noise index" at twice its index, hence 0th node (pythonic zero) will start at 0, first node at 2, second node at 4 etc etc